src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -0,0 +1,369 @@
+        return organizationExternalDatabaseTables[tableId]!!
+    }
+
+    fun getOrganizationExternalDatabaseColumn(columnId: UUID): OrganizationExternalDatabaseColumn {
+        return organizationExternalDatabaseColumns[columnId]!!
+    }
+
+    /*DELETE*/
+    fun deleteOrganizationExternalDatabaseTables(orgId: UUID, tableIds: Set<UUID>) {
+        tableIds.forEach { deleteOrganizationExternalDatabaseTable(orgId, it) }
+    }
+
+    fun deleteOrganizationExternalDatabaseTable(orgId: UUID, tableId: UUID) {
+        organizationExternalDatabaseTables.remove(tableId) //TODO make this a set so we can batch delete, entry processor?
+        aclKeyReservations.release(tableId)
+
+        //delete columns that belonged to this table
+        val belongsToDeletedTable = Predicates.equal("tableId", tableId)
+        val columnsToDelete = organizationExternalDatabaseColumns.values(belongsToDeletedTable)
+        val columnIds = columnsToDelete.map { it.id }.toSet()
+        deleteOrganizationExternalDatabaseColumns(orgId, columnIds)
+
+    }
+
+    fun deleteOrganizationExternalDatabaseColumns(orgId: UUID, columnIds: Set<UUID>) {
+        columnIds.forEach { deleteOrganizationExternalDatabaseColumn(orgId, it) }
+    }
+
+    fun deleteOrganizationExternalDatabaseColumn(orgId: UUID, columnId: UUID) {
+        organizationExternalDatabaseColumns.remove(columnId) //TODO make this a set so we can batch delete, entry processor?
+        aclKeyReservations.release(columnId)
+    }
+
+    /**
+     * Deletes an organization's entire database.
+     * Is called when an organization is deleted.
+     */
+    //TODO break into further helper methods to reduce reuse of code
+    fun deleteOrganizationExternalDatabase(orgId: UUID) {
+        //remove all tables/columns within org
+        val belongsToDeletedDB = Predicates.equal("organizationId", orgId)
+        val tablesToDelete = organizationExternalDatabaseTables.values(belongsToDeletedDB)
+        tablesToDelete.forEach { table ->
+            organizationExternalDatabaseTables.remove(table.id)
+            aclKeyReservations.release(table.id)
+            val belongsToDeletedTable = Predicates.equal("tableId", table.id)
+            val columnsToDelete = organizationExternalDatabaseColumns.values(belongsToDeletedTable)
+            columnsToDelete.forEach { column ->
+                organizationExternalDatabaseColumns.remove(column.id)
+                aclKeyReservations.release(column.id)
+            }
+        }
+
+        //drop db from schema
+        val dbName = PostgresDatabases.buildOrganizationDatabaseName(orgId)
+        assemblerConnectionManager.connect(dbName).use {
+            it.connection.createStatement().use { stmt ->
+                stmt.execute("DROP DATABASE $dbName")
+            }
+        }
+    }
+
+    /*PERMISSIONS*/
+    fun addTrustedUser(orgId: UUID, userPrincipal: Principal, ipAdresses: Set<String>) {
```

Comment:
Shouldn't there be a call for adding trusted role? Or it's not in the requirements for this feature? (We are creating automaticcaly a postgres role for every openlattice role btw)

Reply:
I honestly am not sure yet, I haven't gotten to this piece of the feature yet. I'll look into it though and get back to you!